src/client/options.rs

```diff
@@ -1817,47 +1806,26 @@ impl ConnectionString {
             }
             "authsource" => parts.auth_source = Some(value.to_string()),
             "authmechanismproperties" => {
-                let mut doc = Document::new();
-                let err_func = || {
-                    ErrorKind::InvalidArgument {
-                        message: "improperly formatted authMechanismProperties".to_string(),
-                    }
-                    .into()
-                };
+                let mut properties = Document::new();
```

Comment:
no functional changes were needed here, just did a bit of cleanup to remove the last use of `exclusive_split_at`